crowbar_framework/app/models/network_service.rb

```diff
@@ -323,7 +329,7 @@ def transition(inst, name, state)
 
         # Allocate the bastion network ip for the admin node if a bastion
         # network is defined in the network proposal
-        bastion_net = Chef::DataBag.load "crowbar/bastion_network" rescue nil
+        bastion_net = node_networks["bastion"]
```

Comment:
This is wrong, we need to look at the network role here as we are trying to decide if we should allocate an address on the bastion network.